src/components/RichTextEditor/RichTextEditor.tsx

```diff
@@ -87,6 +66,30 @@ export const RichTextEditor: FC<RichTextEditorProps> = ({
         return observer;
     }, []);
 
+    useEffect(() => {
+        if (!designTokens.heading1) {
```

Comment:
You could do a loop over the existing `textstyles`. This way it would also be it's also dynamic when a new textstyle gets introduced.

```
for (const element of Object.values(TextStyles).filter(
            (textStyle) => textStyle !== TextStyles.ELEMENT_PARAGRAPH,
        )) {
            if (!designTokens[element]) {
                designTokens[element] = defaultDesignTokens[element];
            }
        }
        ```